Tutorial added to Tank Operation activity

ksraj123 opened the pull request:
Fixes Issue #597 
- [x] Should display a welcome message to explain the activity (could reuse the help message describing the activity in Sugarizer activities List view)
- [x] Should explain usage of each button in the toolbar
- [x] Should explain usage of each element in the UI
- [x] Should be localized
- [x] Should respect the Sugarizer UI, as explained in tutorial

- ENGLISH
![tutorial](https://user-images.githubusercontent.com/39027928/75092167-3d012b00-559b-11ea-9ba8-624944ff46f5.gif)

- FRENCH
![tutorial-french](https://user-images.githubusercontent.com/39027928/75094074-ad18ac80-55ad-11ea-8cc2-86324be88f0c.gif)

llaske commented on activities/TankOp.activity/css/activity.css:
```diff
@@ -18,4 +22,132 @@
+}
+
+.tutorial-title {
+	background-color: #808080 !important;
```
Do not mix indentation style in the same file

ksraj123 replied:
Indention Issue Fixed in activity.css file

llaske commented on activities/TankOp.activity/lib/settings.js:
```diff
@@ -10,6 +10,7 @@ define(["webL10n","sugar-web/env"], function (webL10n,env) {
 		env.getEnvironment(function(err, environment) {
 			var defaultLanguage = (typeof chrome != 'undefined' && chrome.app && chrome.app.runtime) ? chrome.i18n.getUILanguage() : navigator.language;
 			var language = environment.user ? environment.user.language : defaultLanguage;
+			console.log(environment.user.language);
```
Remove debug message

ksraj123 replied:
Debug message removed.

llaske commented on activities/TankOp.activity/lib/tutorial.js:
```diff
@@ -0,0 +1,86 @@
+				element: "#fullscreen-button",
+				placement: "bottom",
+				title: l10n.get("TutoFullScreenButtonTitle"),
+				content: l10n.get("TutoFullScreenButtonContent")
```
Do not mix indentation style in the same file

ksraj123 replied:
Indention made uniform across tutorial.js file

llaske commented on activities/TankOp.activity/locale.ini:
Don't do automatic translation, you could use the english localization for all languages

ksraj123 replied:
@llaske Should I use english localization for all titles or should I just use it for only these 4?

s-mag replied:
Use it for all. Native speakers will translate them

ksraj123 replied:
@llaske automatically translated text replaced with english.

@sashr - Thank you for clarification.
> Use it for all. Native speakers will translate them

ksraj123 commented:
@llaske Tutorial added to play screen.
![play mode tutorials](https://user-images.githubusercontent.com/39027928/75098459-8c1b8000-55dc-11ea-96e2-87911c6f5665.gif)

llaske commented on activities/TankOp.activity/images/go-right.svg:
```diff
@@ -43,15 +43,15 @@
      display="block"
      id="go-right"
-     style="fill:#333333">
+     style="fill:#ffffff">
```
You're not supposed to change this. It should be adapted automatically when disable/enable

ksraj123 replied:
Modified files replaced with original svg files. Working as expected now.

llaske commented on activities/TankOp.activity/index.html:
```diff
@@ -42,6 +49,7 @@
 	  <button class="toolbutton" id="activity-button" title="Tank Operation"></button>
 	  <p class="title">TANK OPERATION</p>
 	  <button class="toolbutton pull-right" id="stop-button" title="Stop"></button>
+	  <button class="toolbutton pull-right" id="help-button" title="Tutorial"></button>
```
The help button should be to the left of fullscreen button

ksraj123 replied:
Order Corrected.

llaske commented on activities/TankOp.activity/lib/tutorial.js:
```diff
@@ -0,0 +1,114 @@
+	var tutorial = {};
+
+	tutorial.start = function(play) {
+		var homeSteps = [
```
Still some indentation mix, check your editor:

![image](https://user-images.githubusercontent.com/981841/75098834-3f737d00-55bb-11ea-981f-3aade69ca242.png)

ksraj123 replied:
Indention mix issue in tutorials.js is resolved.

llaske commented:
Good.

Still few issues:

* The arrow to change mission are now invisible

![image](https://user-images.githubusercontent.com/981841/75115817-794b8e80-5662-11ea-94ef-1c591425d87c.png)

I guess it's related to a conflict with existing icons. In that case, just duplicate icons to avoid the conflict.

* Miss a popup to explain missions completed zone
* I suggest to set position for popup on start button at top instead of bottom to avoid scrollbar
* I suggest you to move popup for Credits, Fullscreen and Stop at the end of the array
* I suggest to add a popup in Play mode to explain the meaning of LCD display

ksraj123 commented:
@llaske Modifications made as per suggestions.

- Fixed Invisible mission changing icon issue
- Added a popup for mission completed zone
- Changed the position for start button popup to top
- Moved the popup for credits, full screen and stop to the last of the array
- Added a pop up for the canvas in play mode
- Added popups for wave count and score
- Modified the popup for virtual keyboard to select the entire keyboard not just first three buttons
- Modified the popup for change mission to select the entire thing not just right button

Home Screen
![homeModifier](https://user-images.githubusercontent.com/39027928/75121938-fc9acd80-56be-11ea-83aa-88e42f906a2f.gif)

Play Screen
![playModified](https://user-images.githubusercontent.com/39027928/75121777-3b2f8880-56bd-11ea-9627-2f7755d6d346.gif)

llaske commented:
Good job. Thanks.
